Update Contact.rst

robinsonmm opened the pull request:
Integrate etiquette and other elements from old Support guideliens #1099 from ros2_documentation.

robinsonmm committed:
Update Contact.rst

Integrate etiquette and other elements from old Support guideliens #1099 from ros2_documentation.

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,30 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
 
 Before opening an issue, check if other users have reported similar issues by searching across the ros2 and ament GitHub organizations: `example search query <https://github.com/search?q=user%3Aros2+user%3Aament+turtlesim&type=Issues>`__.
 
+Next, check `ROS Answers <https://answers.ros.org/>`__ to see if someone else has asked your question or reported your issue. And try a general search you may also find results from `ROS Discourse <https://discourse.ros.org/>`__ the  `ros-users mailing list archive <http://lists.ros.org/lurker/list/ros-users.html>`__ and other sites.
+
```
One sentence per line please.

That said, I'm not sure that second sentence is that helpful.  The ros-user mailing list is going away, so it won't contain very much anymore.  And the general advice of "do a Google search" is good, but I think should be mostly self explanatory.

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,30 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
 When filing an issue, please make sure to:
 
-* Include enough information for another person to understand the issue.
+* Include enough information for another person to understand the issue. Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong. If following a Tutorial or online instructions provide a link to the specific instructions. 
```
```suggestion
* Include enough information for another person to understand the issue. Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong. If following a tutorial or online instructions provide a link to the specific instructions. 
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,30 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
+  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it). 
+* Any warnings or errors. Cut and paste them directly from the terminal window to which they were printed. Please do not re-type or include a screenshot.  
 * In case of a bug consider providing a `short, self contained, correct (compilable), example <http://sscce.org/>`__.
+* When discussing any compiling/linking/installation issues, also provide gcc version
```
```suggestion
* When discussing any compiling/linking/installation issues, also provide the compiler version
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,30 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
+* Ogre.log for rviz, if possible (run with rviz -l)
+* Bag files and code samples that can reproduce the problem
+* Gifs or movies to demonstrate the problem
+Always tag your questions appropriately. You must tag at least the rosdistro version. Format code snippets. To make your code snippets and error messages human readable, format them correctly (mark the code and press Ctrl-k).
```
The first sentence isn't actually true for issues on GitHub; we don't have tags for the individual distributions there.  The third and fourth sentences don't seem to be correct either.  I think it might make sense to just remove this whole line.

robinsonmm replied:
I have removed that line as noted. Nice catch.

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +87,27 @@ Participating in these discussions is an important way to have a say on how diff
+Etiquette
+----------
+
+Assume 'good faith': It's easy to mis-interpret the meaning or tone of comments on the internet. Assuming good faith gives the benefit of the doubt to those trying to help you, avoiding: insulting well meaning community members, and poisoning the mood. Assuming 'good faith' when responding almost always works better even if the original response was not in fact in good faith.
```
One sentence per line here (and in the rest of this section).

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +87,27 @@ Participating in these discussions is an important way to have a say on how diff
+
+On https://answers.ros.org you can edit your question to provide more details. The more details that you include in your question the easier it is for others to help you find your solution which makes it more likely for you to get a response.
+
+It's considered bad form to list your personal deadlines; Community members answering questions also have them.
```
```suggestion
It's considered bad form to list your personal deadlines; community members answering questions also have them.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +87,27 @@ Participating in these discussions is an important way to have a say on how diff
+
+* For commercial posts, see also `this discussion <https://discourse.ros.org/t/sponsorship-notation-in-posts-on-ros-org/2078>`_.
+
+Minimize references to content behind pay walls. The content posted on the wiki and in Ask should "generally" be free and open to all users. Links to content behind pay walls such as private journal articles, text books, and paid news websites, while helpful and relevant, may not be accessible to all users. Where possible primary sources should be free and open with paid content playing a support ...
```
So we don't have "Ask" or "the wiki" in ROS 2.  Probably we should change this to "Discourse" and "Answers".

clalancette reviewed (changes requested):
Thanks for taking this on!  I've left a bunch of small things to cleanup in here.

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update Contact.rst

This is in response to the review by clalancette.

robinsonmm reviewed (commented):
@clalancette I think I got your feedback incorporated. Please feel free to note if I have other things that are a bit clunky.

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,34 @@ Participating in these discussions is an important way to have a say on how diff
+Try to pick the forum which you think matches best and ask there. 
+If you are referred to a new forum, provide a link to the old discussion.
+
+On https://answers.ros.org you can edit your question to provide more details. The more details that you include in your question the easier it is for others to help you find your solution which makes it more likely for you to get a response.
```
```suggestion
On https://answers.ros.org you can edit your question to provide more details.
The more details that you include in your question the easier it is for others to help you find your solution which makes it more likely for you to get a response.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,34 @@ Participating in these discussions is an important way to have a say on how diff
+
+It's considered bad form to list your personal deadlines; community members answering questions also have them.
+
+Do not beg for help. If there is someone willing and able to help with your problem, you usually get a response. Asking for faster answers will mostly have a negative effect.
```
```suggestion
Do not beg for help. If there is someone willing and able to help with your problem, you usually get a response.
Asking for faster answers will mostly have a negative effect.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,34 @@ Participating in these discussions is an important way to have a say on how diff
+
+Do not beg for help. If there is someone willing and able to help with your problem, you usually get a response. Asking for faster answers will mostly have a negative effect.
+
+Do not add unrelated content to posts. The content of posts should be focused on the topic at hand and not include unrelated content. Content, links, and images unrelated to the topic are considered spam.
```
```suggestion
Do not add unrelated content to posts.
The content of posts should be focused on the topic at hand and not include unrelated content.
Content, links, and images unrelated to the topic are considered spam.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,34 @@ Participating in these discussions is an important way to have a say on how diff
+
+* For commercial posts, see also `this discussion <https://discourse.ros.org/t/sponsorship-notation-in-posts-on-ros-org/2078>`_.
+
+Minimize references to content behind pay walls. The content posted on `ROS Discourse <https://discourse.ros.org/>` and `ROS Answers <https://answers.ros.org/>` should "generally" be free and open to all users. Links to content behind pay walls such as private journal articles, text books, and paid news websites, while helpful and relevant, may not be accessible to all users. Where possible prima ...
```
```suggestion
Minimize references to content behind pay walls.
The content posted on `ROS Discourse <https://discourse.ros.org/>` and `ROS Answers <https://answers.ros.org/>` should "generally" be free and open to all users.
Links to content behind pay walls such as private journal articles, text books, and paid news websites, while helpful and relevant, may not be accessible to all users.
Where possible primary sources should be free and open with paid content playing a supporting role.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,34 @@ Participating in these discussions is an important way to have a say on how diff
+
+Minimize references to content behind pay walls. The content posted on `ROS Discourse <https://discourse.ros.org/>` and `ROS Answers <https://answers.ros.org/>` should "generally" be free and open to all users. Links to content behind pay walls such as private journal articles, text books, and paid news websites, while helpful and relevant, may not be accessible to all users. Where possible prima ...
+
+Single link posts are to be avoided. Generally speaking, posting a single link answer is less helpful and can be easily confused with spam. Moreover, links may degrade over time or be replaced. Paraphrasing a link's content along with some contextual information and attribution is often much more helpful.
```
```suggestion
Single link posts are to be avoided.
Generally speaking, posting a single link answer is less helpful and can be easily confused with spam.
Moreover, links may degrade over time or be replaced.
Paraphrasing a link's content along with some contextual information and attribution is often much more helpful.
```

clalancette reviewed (changes requested):
Just a few more fixes for one-sentence-per-line, then this will be good to go.

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm commented:
I have committed your requested changes. Thanks for your patience with the formatting/one line-one sentence.

robinsonmm commented:
I think I got them. Thanks for making the commits.

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,44 @@ Participating in these discussions is an important way to have a say on how diff
+
+Please don't send your question more than once: The question was seen. 
+If you didn't get a response then likely nobody has had time to answer you. 
+Alternatively, it could be that nobody knows the answer. In any case, sending it again is poor form and akin to shouting and is likely to aggravate a large number of people. 
```
```suggestion
Alternatively, it could be that nobody knows the answer.
In any case, sending it again is poor form and akin to shouting and is likely to aggravate a large number of people. 
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,44 @@ Participating in these discussions is an important way to have a say on how diff
+
+It's considered bad form to list your personal deadlines; community members answering questions also have them.
+
+Do not beg for help. If there is someone willing and able to help with your problem, you usually get a response.
```
```suggestion
Do not beg for help.
If there is someone willing and able to help with your problem, you usually get a response.
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,32 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
 When filing an issue, please make sure to:
 
-* Include enough information for another person to understand the issue.
+* Include enough information for another person to understand the issue. 
+Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong. 
```
```suggestion
* Include enough information for another person to understand the issue.

Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong. 
```

(otherwise, sphinx warns while building)

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,32 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
 
-* Include enough information for another person to understand the issue.
+* Include enough information for another person to understand the issue. 
+Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong. 
+If following a tutorial or online instructions provide a link to the specific instructions. 
+* Use a descriptive headline or subject line. Bad: "rviz doesn't work". Good: "Rviz crashing looking for missing .so after latest apt update"
 * Include information about the exact platform, software, versions, and environment relevant to the problem.
-  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it).
+  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it). 
```
Sphinx is also warning about this, though I don't quite know why.

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +89,44 @@ Participating in these discussions is an important way to have a say on how diff
+* For commercial posts, see also `this discussion <https://discourse.ros.org/t/sponsorship-notation-in-posts-on-ros-org/2078>`_.
+
+Minimize references to content behind pay walls.
+The content posted on `ROS Discourse <https://discourse.ros.org/>` and `ROS Answers <https://answers.ros.org/>` should "generally" be free and open to all users.
```
```suggestion
The content posted on `ROS Discourse <https://discourse.ros.org/>`__ and `ROS Answers <https://answers.ros.org/>`__ should "generally" be free and open to all users.
```

clalancette reviewed (changes requested):
After running this through sphinx, I saw a few more errors.  I think if you take my suggestions here, that will fix the rest of them.  Sorry for all of the back and forth; we are getting closer :).

gavanderhoorn commented:
Friendly ping @robinsonmm

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm commented:
I have accepted the commits. Thanks for the follow up/sorry for the delay in response.

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,33 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
 
 Before opening an issue, check if other users have reported similar issues by searching across the ros2 and ament GitHub organizations: `example search query <https://github.com/search?q=user%3Aros2+user%3Aament+turtlesim&type=Issues>`__.
 
+Next, check `ROS Answers <https://answers.ros.org/>`__ to see if someone else has asked your question or reported your issue. 
```
```suggestion
Next, check `ROS Answers <https://answers.ros.org/>`__ to see if someone else has asked your question or reported your issue.
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,33 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
 
 * Include enough information for another person to understand the issue.
+
+Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong. 
```
```suggestion
Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong.
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,33 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
 * Include information about the exact platform, software, versions, and environment relevant to the problem.
-  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it).
+  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it). 
+* Any warnings or errors. Cut and paste them directly from the terminal window to which they were printed. Please do not re-type or include a screenshot.  
```
```suggestion
* Any warnings or errors. Cut and paste them directly from the terminal window to which they were printed. Please do not re-type or include a screenshot.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +90,46 @@ Participating in these discussions is an important way to have a say on how diff
+Etiquette
+----------
+
+Assume 'good faith': It's easy to mis-interpret the meaning or tone of comments on the internet. 
```
```suggestion
Assume 'good faith': It's easy to mis-interpret the meaning or tone of comments on the internet.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +90,46 @@ Participating in these discussions is an important way to have a say on how diff
+----------
+
+Assume 'good faith': It's easy to mis-interpret the meaning or tone of comments on the internet. 
+Assuming good faith gives the benefit of the doubt to those trying to help you, avoiding: insulting well meaning community members, and poisoning the mood. 
```
```suggestion
Assuming good faith gives the benefit of the doubt to those trying to help you, avoiding: insulting well meaning community members, and poisoning the mood.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +90,46 @@ Participating in these discussions is an important way to have a say on how diff
+Assuming good faith gives the benefit of the doubt to those trying to help you, avoiding: insulting well meaning community members, and poisoning the mood. 
+Assuming 'good faith' when responding almost always works better even if the original response was not in fact in good faith.
+
+Please don't send your question more than once: The question was seen. 
```
```suggestion
Please don't send your question more than once: The question was seen.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +90,46 @@ Participating in these discussions is an important way to have a say on how diff
+Assuming 'good faith' when responding almost always works better even if the original response was not in fact in good faith.
+
+Please don't send your question more than once: The question was seen. 
+If you didn't get a response then likely nobody has had time to answer you. 
```
```suggestion
If you didn't get a response then likely nobody has had time to answer you.
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +90,46 @@ Participating in these discussions is an important way to have a say on how diff
+Alternatively, it could be that nobody knows the answer.
+In any case, sending it again is poor form and akin to shouting and is likely to aggravate a large number of people. 
+This also applies to crossposting. 
+Try to pick the forum which you think matches best and ask there. 
```
```suggestion
Try to pick the forum which you think matches best and ask there.
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,33 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
+
+Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong. 
+If following a tutorial or online instructions provide a link to the specific instructions. 
+* Use a descriptive headline or subject line. Bad: "rviz doesn't work". Good: "Rviz crashing looking for missing .so after latest apt update"
```
```suggestion
If following a tutorial or online instructions provide a link to the specific instructions. 

* Use a descriptive headline or subject line. Bad: "rviz doesn't work". Good: "Rviz crashing looking for missing .so after latest apt update"
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,33 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
+* Use a descriptive headline or subject line. Bad: "rviz doesn't work". Good: "Rviz crashing looking for missing .so after latest apt update"
 * Include information about the exact platform, software, versions, and environment relevant to the problem.
-  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it).
+  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it). 
```
(I can't leave a suggestion across a deleted line here)

Sphinx is still complaining about this.  I think what you have to do here is put the "This includes..." sentence on the same line as the previous one "Include information about...".  That should fix this error.

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +90,46 @@ Participating in these discussions is an important way to have a say on how diff
+The content of posts should be focused on the topic at hand and not include unrelated content.
+Content, links, and images unrelated to the topic are considered spam.
+
+* For commercial posts, see also `this discussion <https://discourse.ros.org/t/sponsorship-notation-in-posts-on-ros-org/2078>`_.
```
```suggestion
For commercial posts, see also `this discussion <https://discourse.ros.org/t/sponsorship-notation-in-posts-on-ros-org/2078>`_.
```

(the lone bullet doesn't make much sense here)

clalancette reviewed (changes requested):
A few more things to change.  Sphinx is still complaining about some things, and reading the page points out that some of these things didn't render properly.

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update source/Contact.rst

Co-authored-by: Chris Lalancette <clalancette@gmail.com>

robinsonmm committed:
Update Contact.rst

I edited per your suggestion on that deleted line. I accepted the rest of the suggestions. Hopefully good to go?

robinsonmm reviewed (commented):
I committed the suggestion on the deleted line and accepted the suggestions. Hopefully Sphinx agrees. Thanks!

clalancette reviewed (approved):
This looks good now, thanks for all of the iteration.

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,33 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
-  This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it).
+
+Describe exactly what you were doing or are trying to do, and exactly what, if anything, went wrong.
+If following a tutorial or online instructions provide a link to the specific instructions. 
```
```suggestion
If following a tutorial or online instructions provide a link to the specific instructions.
```

clalancette commented on source/Contact.rst:
```diff
@@ -40,15 +40,33 @@ You can search for individual ROS 2 repositories on `ROS 2's GitHub <https://git
+If following a tutorial or online instructions provide a link to the specific instructions. 
+
+* Use a descriptive headline or subject line. Bad: "rviz doesn't work". Good: "Rviz crashing looking for missing .so after latest apt update"
+* Include information about the exact platform, software, versions, and environment relevant to the problem. This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it).   
```
```suggestion
* Include information about the exact platform, software, versions, and environment relevant to the problem. This includes how you installed the software (from binaries or from source) and which ROS middleware/DDS vendor you are using (if you know it).
```

clalancette commented on source/Contact.rst:
```diff
@@ -72,6 +90,46 @@ Participating in these discussions is an important way to have a say on how diff
+Please don't send your question more than once: The question was seen.
+If you didn't get a response then likely nobody has had time to answer you.
+Alternatively, it could be that nobody knows the answer.
+In any case, sending it again is poor form and akin to shouting and is likely to aggravate a large number of people. 
+This also applies to crossposting. 
```
```suggestion
In any case, sending it again is poor form and akin to shouting and is likely to aggravate a large number of people.
This also applies to crossposting.
```

clalancette reviewed (changes requested):
CI is unhappy because of trailing whitespace.  With these last changes applied, it should be good to go.

clalancette commented:
Closing in favor of #2939, so we can get this in.
